src/scanners/machine_uuid_scanner.go

```diff
@@ -36,24 +53,24 @@ func readSystemUUID() *strfmt.UUID {
 	return &ret
 }
 
-func readMotherboardSerial() *strfmt.UUID {
-	basedboard, err := ghw.Baseboard()
+func readMotherboardSerial(d Dependencies) *strfmt.UUID {
+	basedboard, err := d.Baseboard()
 	if err != nil {
 		log.Warnf("Could not find motherboard serial number: %s", err.Error())
 		return nil
 	}
 	value := basedboard.SerialNumber
-	if value == "" || value == ghw.UNKNOWN {
+	if value == "" || value == ghw.UNKNOWN || value == VmwareDefualtSerial {
 		log.Warn("Could not find motherboard serial number")
 		return nil
 	}
 	return md5GenerateUUID(value)
 }
 
-func ReadId() *strfmt.UUID {
-	ret := readMotherboardSerial()
+func ReadId(d Dependencies) *strfmt.UUID {
+	ret := readMotherboardSerial(d)
 	if ret == nil {
-		ret = readSystemUUID()
+		ret = readSystemUUID(d)
```

Comment:
We still might get all hosts with the same DefaultUUID.
Why don't not generate some random UUID instead of the default (0000-...)?

Reply:
0 uuid means that something went wrong.  I prefer to fail on it